packages/devtools_app/lib/src/config_specific/import_export/import_export.dart

```diff
@@ -59,15 +59,17 @@ class ImportController {
     }
     previousImportTime = now;
 
-    final isDevToolsSnapshot = json[devToolsSnapshotKey];
+    final isDevToolsSnapshot =
+        json is Map<String, dynamic> && json[devToolsSnapshotKey];
     if (isDevToolsSnapshot == null || !isDevToolsSnapshot) {
       _notifications.push(nonDevToolsFileMessage);
       return;
     }
 
+    final devToolsSnapshot = json as Map<String, dynamic>;
     // TODO(kenz): support imports for more than one screen at a time.
-    final activeScreenId = json[activeScreenIdKey];
-    offlineDataJson = json;
+    final activeScreenId = devToolsSnapshot[activeScreenIdKey];
+    offlineDataJson = devToolsSnapshot;
```

Comment:
Are the additions to this file just to make sure we don't accept instruction sizes/v8 snapshot for offline mode?

Reply:
This file uses the `DevToolsJsonFile` object as well. So we need to cast to a map so that we can check for the active screen id. If it is not a map, we know it's not an offline import and we will show the `nonDevToolsFileMessage` message.